### PR TITLE
Keystore generation

### PR DESCRIPTION
**Pull request by impul**
### Description
Add Keystore file generation
Add Keystore decoding and encoding
Add Keystore decryption and encryption
Update pods before build on Travis

### CODE REVIEW

File: HDWalletKit/Keystore/KeystoreV3.swift [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        guard let calculatedMac = Data.fromHex(keystoreParams.crypto.mac),
+              mac.constantTimeComparisonTo(calculatedMac) else {return nil}
+        let decryptionKey = derivedKey[0...15]
+        guard let IV = Data.fromHex(keystoreParams.crypto.cipherparams.iv) else {return nil}
+        guard let aesCipher = try? AES(key: decryptionKey.bytes, blockMode: CTR(iv: IV.bytes), padding: .noPadding) else {return nil}
+        guard let decryptedPK:Array<UInt8> = try? aesCipher.decrypt(cipherText.bytes) else { return nil }
+        return Data(bytes:decryptedPK)
+    }
+    
+    private func scrypt(password: String, salt: Data, length: Int, N: Int, R: Int, P: Int) -> Data? {
```
Comment (kozlov-oleksandr) [inline HDWalletKit/Keystore/KeystoreV3.swift]: @impul scrypt?

File: HDWalletKit/Keystore/KeystoreV3.swift [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        let keystoreparams = KeystoreParamsV3(crypto: crypto, id: UUID().uuidString.lowercased(), version: 3)
+        self.keystoreParams = keystoreparams
+    }
+}
+
+
+
+
+
+
```
Comment (kozlov-oleksandr) [inline HDWalletKit/Keystore/KeystoreV3.swift]: @impul redundant blank lines

**Review by kozlov-oleksandr — CHANGES_REQUESTED** [review]

File: HDWalletKit/Keystore/KeystoreV3.swift (hunk 2 above)
Comment (impul) [reply HDWalletKit/Keystore/KeystoreV3.swift]: @kozlov-oleksandr, done.

File: HDWalletKit/Keystore/KeystoreV3.swift (hunk 1 above)
Comment (impul) [reply HDWalletKit/Keystore/KeystoreV3.swift]: @kozlov-oleksandr, done.

**Review by kozlov-oleksandr — APPROVED** [review]

**Pull request merged**